internal/fsutils/fsutils.go

```diff
@@ -91,3 +92,42 @@ func EvalSymlinks(path string) (string, error) {
 
 	return er.path, er.err
 }
+
+func toRegexp(pattern string) string {
+	// code from https://github.com/guillermo/doubleglob/blob/main/double_glob.go
+	replaces := regexp.MustCompile(`(\.)|(\*\*/)|(\*)|([^/\*.]+)|(/)`)
```

Comment:
One more thing, why do we use groups here? As far as I can see, we just match whether this regexp fits our string or not.